src/charm.py

```diff
@@ -230,7 +230,7 @@ def _configured(self) -> bool:
         """
         return bool(self.config.get("git_repo"))
 
-    def _exec_sync_repo(self) -> Tuple[str, str]:
+    def _exec_sync_repo(self) -> Tuple[str, Optional[str]]:
```

Comment:
It seems that keeping `Tuple[str, str]` and changing to `return stdout, stderr or ""` would be cleaner to consume.